### PR TITLE
Refactor dependencies handling to keep necessary runtime dependencies automatically

### LINKED ISSUE #300: Python 3.7 Alpine 3.8
Since Alpine 3.8 has been released https://alpinelinux.org/posts/Alpine-3.8.0-released.html, it would be great to have python:3.7-alpine3.8

It contains some nice goodies, like newer OpenSSL.

Probably we would need to wait some days to see an official Alpine 3.8 image.

### PR DESCRIPTION
**Pull request by tianon**
(accounting for "tkinter" and "libssl" edge cases appropriately)

Closes #309
Closes #306
Closes #303
Closes #283

(combined/refactored)

Closes #300

### CODE REVIEW

File: Dockerfile-slim.template [hunk 1]
```diff
@@ -72,6 +66,13 @@ RUN set -ex \
 	\
 	&& apt-mark auto '.*' > /dev/null \
 	&& apt-mark manual $savedAptMark \
+	&& find /usr/local -type f -executable -not \( -name '*tkinter*' \) -exec ldd '{}' ';' \
```
Comment (tianon) [inline Dockerfile-slim.template]: See https://github.com/docker-library/python/issues/122 and https://github.com/docker-library/python/pull/130 (and https://github.com/docker-library/python/pull/283) for background on `tk` / `tkinter`.

**Comment by tianon** [comment]
Added https://github.com/docker-library/php/pull/666 too, while we're at it.

**Comment by tianon** [comment]
Also more Alpine 3.8 (all the Alpine 3.8 possible, in fact).

File: 3.4/alpine3.8/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+ENV PATH /usr/local/bin:$PATH
+
+# http://bugs.python.org/issue19846
+# > At the moment, setting "LANG=C" on a Linux system *fundamentally breaks Python 3*, and that's not OK.
+ENV LANG C.UTF-8
+
+# install ca-certificates so that HTTPS works consistently
+# the other runtime dependencies for Python are installed later
+RUN apk add --no-cache ca-certificates
+
```
Comment (tao12345666333) [inline 3.4/alpine3.8/Dockerfile]: Should we remove ca-certificates? They are included on Alpine 3.8.

**Pull request merged**